src/test/java/com/snowflake/kafka/connector/records/RecordContentTest.java

```diff
@@ -125,103 +130,98 @@ public void test() throws IOException {
             "{\"int8\":12,\"int16\":12,\"int32\":12,\"int64\":12,\"float32\":12.2,\"float64\":12.2,\"boolean\":true,\"string\":\"foo\",\"bytes\":\"Zm9v\",\"array\":[\"a\",\"b\",\"c\"],\"map\":{\"field\":1},\"mapNonStringKeys\":[[1,1]]}");
   }
 
-  @Test(expected = SnowflakeKafkaConnectorException.class)
-  public void testEmptyValueDisabledTombstone() {
+  @ParameterizedTest
+  @MethodSource("disabledTombstoneSource")
+  public void recordService_getProcessedRecordForSnowpipe_whenDisabledTombstone_throwException(
+      String testCase, SinkRecord record) {
+    // given
     RecordService service = new RecordService();
     service.setBehaviorOnNullValues(SnowflakeSinkConnectorConfig.BehaviorOnNullValues.IGNORE);
 
-    SinkRecord record =
-        new SinkRecord(topic, partition, null, null, Schema.STRING_SCHEMA, null, partition);
-    service.getProcessedRecordForSnowpipe(record);
+    // expect
+    Assertions.assertThrows(
+        SnowflakeKafkaConnectorException.class,
+        () -> service.getProcessedRecordForSnowpipe(record));
   }
 
-  @Test(expected = SnowflakeKafkaConnectorException.class)
-  public void testEmptyValueSchemaDisabledTombstone() throws IOException {
+  public static Stream<Arguments> disabledTombstoneSource() throws JsonProcessingException {
     JsonNode data = mapper.readTree("{\"name\":123}");
     SnowflakeRecordContent content = new SnowflakeRecordContent(data);
-    RecordService service = new RecordService();
-    service.setBehaviorOnNullValues(SnowflakeSinkConnectorConfig.BehaviorOnNullValues.IGNORE);
-
-    SinkRecord record = new SinkRecord(topic, partition, null, null, null, content, partition);
-    service.getProcessedRecordForSnowpipe(record);
+    return Stream.of(
+        Arguments.of(
+            "empty value with schema",
+            new SinkRecord(topic, partition, null, null, Schema.STRING_SCHEMA, null, partition)),
+        Arguments.of(
+            "not empty value without schema",
+            new SinkRecord(topic, partition, null, null, null, content, partition)));
   }
 
-  @Test(expected = SnowflakeKafkaConnectorException.class)
-  public void testWrongValueSchema() throws IOException {
-    JsonNode data = mapper.readTree("{\"name\":123}");
-    SnowflakeRecordContent content = new SnowflakeRecordContent(data);
+  @ParameterizedTest
+  @MethodSource("invalidSchemaSource")
+  public void recordService_getProcessedRecordForSnowpipe_whenInvalidSchema_throwException(
+      String testCase, Schema schema, Object value) {
```

Comment:
Let's use `Named` arguments instead of passing a test case name. See [a stackoverflow thread](https://stackoverflow.com/questions/57892989/generating-display-names-for-parameterizedtest-in-junit-5).

Reply:
done